### PR TITLE
editoast: generate errors for out of range slopes or curves

### LINKED ISSUE #3458: Slopes and Curves range are not check against the track section length
### What happened?

When we add a slope/curve with an invalid range (end position higher than the track length). No errors are found by editoast. However, it makes the not loadable.

### What did you expect to happen?

An `out_of_range` error should be triggered by editoast.

### How can we reproduce it (as minimally and precisely as possible)?

1. Patch a track with a new slope/curve with an end value higher than the track length
2. No error is triggered.

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Nixos v22.11
- **Env**: Local / Dev

### OSRD version (top right corner `Account` button > `Informations`)

899cb897

### PR DESCRIPTION
**Pull request by flomonster**
close #3458

### CODE REVIEW

File: editoast/Cargo.toml [hunk 1]
```diff
@@ -48,3 +48,4 @@ editoast_derive = { path = "./editoast_derive" }
 mvt = "0.8.1"
 pointy = "0.4.0"
 futures = "0.3.26"
+ntest = "0.9.0"
```
Comment (bloussou) [inline editoast/Cargo.toml:53]: Shouldn't it be under [`dev-dependencies`](https://doc.rust-lang.org/rust-by-example/testing/dev_dependencies.html) ?

File: editoast/src/generated_data/error/track_sections.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::InfraError;
+    use super::{check_curve_out_of_range, check_slope_out_of_range};
+    use crate::infra_cache::tests::{
+        create_small_infra_cache, create_track_endpoint, create_track_link_cache,
+        create_track_section_cache,
+    };
```
Comment (younesschrifi) [inline editoast/src/generated_data/error/track_sections.rs]:
```suggestion
        use crate::infra_cache::tests::{create_small_infra_cache, create_track_section_cache};

```

File: editoast/src/generated_data/error/track_sections.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    use crate::infra_cache::Graph;
+    use crate::schema::{Curve, Endpoint, OSRDObject, ObjectRef, ObjectType, Slope, TrackEndpoint};
```
Comment (younesschrifi) [inline editoast/src/generated_data/error/track_sections.rs]:
```suggestion
        use crate::schema::{Curve, Slope};
```

File: editoast/src/generated_data/error/track_sections.rs (hunk 2 above)
Comment (younesschrifi) [reply editoast/src/generated_data/error/track_sections.rs]: unused import

File: editoast/src/generated_data/error/track_sections.rs (hunk 3 above)
Comment (younesschrifi) [reply editoast/src/generated_data/error/track_sections.rs]: unused import

**Review by younesschrifi — CHANGES_REQUESTED** [review]

**Review by younesschrifi — APPROVED** [review]

**Comment by Tguisnet** [comment]
Good PR, nothing to change to be merged, however : 
Currently the user can't select the begin and the end of slopes/curves.
- For me we could choose to stay like this, but we should assure that the ranges are correct (so we need to fix this bug : https://github.com/DGEXSolutions/osrd/issues/3488) and in this case, this error become useless.
- We choose to let the user a total control of the begin and the end

**Comment by flomonster** [comment]
This error won't be useless. Using the editor is not the only way to add infra. A user can upload his own infra. In this case, we must check everything.

**Review by Tguisnet — APPROVED** [review]

**Pull request merged**